packages/auth/src/providers/cognito/apis/signInWithUserPassword.ts

```diff
@@ -0,0 +1,79 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { AuthValidationErrorCode } from '../../../errors/types/validation';
+import { assertServiceError } from '../../../errors/utils/assertServiceError';
+import { assertValidationError } from '../../../errors/utils/assertValidationError';
+import { AuthSignInStep, SignInRequest } from '../../../types';
+import { CognitoSignInOptions } from '../types/options/CognitoSignInOptions';
+import { setActiveSignInSession } from '../utils/activeSignInSession';
+import {
+	ChallengeName,
+	ChallengeParameters,
+} from '../utils/clients/types/models';
+import {
+	getSignInResult,
+	getSignInResultFromError,
+	handleUserPasswordAuthFlow,
+} from '../utils/signInHelpers';
+import { InitiateAuthException } from '../types/errors/service';
+import { Amplify } from '@aws-amplify/core';
+
+/**
+ * Signs a user in using USER_PASSWORD_AUTH AuthFlowType
+ *
+ * @param signInRequest - The SignInRequest object
+ * @returns AuthSignInResult
+ * @throws service: {@link InitiateAuthException } -
+ * Cognito service error thrown during the sign-in process.
+ * @throws validation: {@link AuthValidationErrorCode  } - Validation errors thrown either username or password
+ *  are not defined.
+ *
+ * TODO: add config errors
+ */
+export async function signInWithUserPassword(
+	signInRequest: SignInRequest<CognitoSignInOptions>
+) {
+	const { username, password } = signInRequest;
+	const config = Amplify.config;
```

Comment:
Since the `clientMetadata` is the only thing we're using out of config, maybe we can just grab that from the start:

```suggestion
	const clientMetadata = Amplify.config.clientMetadata;
```